### ISSUE TITLE
Update dependencies of React package & Improve Docs

### ISSUE THREAD

**Issue by deadcoder0904** [electron-userland/electron-forge-templates#58]
* [x] I have read the [contribution documentation](https://github.com/electron-userland/electron-forge-templates/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

Update the dependencies of React package its so old. Also, the docs aren't proper. 

After I tried `electron-forge init . --template=react`, it gave me an error saying Install `npm I -g @electron-forge-template-react` & when I tried it, it gave me an error saying it does not exist.

So it should've been `npm i -g electron-forge-template-react` without the `@`

I think you should set up GreenKeeper bot to auto-update packages without requiring more manual work. Also, electron version is old. Now I know you might not wanna use Beta v3 but again it would be better if it can be users decision. Personally, `electron-forge` aims to be simpler & it really is but I think packages need to be updated otherwise its more manual work than setting up `electron-webpack`

**Comment by MarshallOfSound** [comment]
@deadcoder0904 From the sounds of it you jumped into Electron Forge v6 (the version that is currently being built and is still in beta) and then tried to use stuff that hasn't been set up yet.

For instance, this templates repository hasn't been migrated to use v6 and v6 doesn't have real support for templates yet.

See the [Giant Warning](https://github.com/electron-userland/electron-forge#rotating_light-construction-warning-construction-rotating_light) on the `electron-forge` README.

**Comment by MarshallOfSound** [comment]
RE auto updating dependencies, green keeper has historically annoyed me more than it has helped.  The v6 rewrite of templates won't be framework specific anyway, rather build tool templates that you can then use with whatever framework you want.

E.g. There would be a `webpack` template and a `parcel` template, we may have examples for popular frameworks like React but the ethos of templates will change with v6 to be less opinionated and more helpful 👍

**Comment by deadcoder0904** [comment]
Nice awesome. Okay I will try v5 of `electron-forge` & see if it works ;)

I actually tried v6 because you know nowadays every beta version is stable (React, RN, Electron, etc.)😂 & also docs are looking better

**Comment by MarshallOfSound** [comment]
> Electron

Uh, be warned, Electron Beta is explicitly called beta.  It's not "stable" out of the gate, we have these beta cycles to fix new crashes and bugs that appear.  Beta 100% means beta 😄

**Comment by deadcoder0904** [comment]
So I tried v5 & it works out of the box, thank you. But, eslint is shouting. 

Also, I guess most people prefer `.js` as an extension for React rather than `.jsx`. So I did [this](https://stackoverflow.com/a/43031230) for that. Can you make it default❓

Also, it might be a good idea to make Prettier a default or either an opt-in. What do you think❓

**Comment by MarshallOfSound** [comment]
It's worth noting that Electron Forge's templates **were never intended to be boilerplates**.  In fact that's something I steered well clear of, boilerplates are over architected, over opinionated massive piles of code that normally makes no sense to anyone.

To this end in v6 the default linting setup (`eslint` / `standard` / whatever) has been removed from forge.  We want to let users do whatever they want to do easily, but also not make choices for them.  If you don't want to use prettier, don't... If you want to use `.js` instead of `.jsx`, that's fine, go for it.

The goal of `v6` was to empower users to do whatever they want, and be less opinionated by default.  See the plugin system and such for examples of this effort.

TLDR: Linting is not part of core forge anymore, so making changes to these templates for things like eslint configs or adding prettier make little sense

**Comment by deadcoder0904** [comment]
Nice amazing. Okay one last question, my JSX isn't getting code-highlighting❓

Any ETA on v6 release like a month or 6 months❓
